lib/bolt/inventory.rb

```diff
@@ -243,5 +253,39 @@ def set_facts(tname, hash)
       end
     end
     private :set_facts
+
+    def add_node(grp, target, group_name, track = { 'all' => nil })
+      grp.groups.each do |g|
+        track[g.name] = { name: grp.name, data: grp }
+        add_node(g, target, group_name, track) if g.groups.any?
+        next unless g.name == group_name
```

Comment:
It seems like this will fail if you try to add a node to "all" (or the top group)?

I suspect we need to refactor this to check if they current group is the desired group outside the look and then recurse on all children. I also think that will be more readable than next unless.